index.js

```diff
@@ -20,7 +22,7 @@ var requirejs = require('requirejs');
 module.exports = function(options) {
   // If you want to supply a preparsed config object without a configPath, then
   // we have no way of knowing what the baseUrl should be.
-  var configPath = options.configPath || options.filename;
+  var configPath = options.configPath;
```

Comment:
I did some changes in how configPath is set so it's easier to follow what's happening.